### PR TITLE
Iiwa 187

### PR DESCRIPTION
**Pull request by mcdowellalex**

### CODE REVIEW

File: src/frontend-admin/src/components/pages/DashboardPage/DashboardPage.tsx [hunk 1]
```diff
@@ -120,15 +121,15 @@ export const DashboardPage: React.FC<IDashboardPageProps> = props => {
             deptList.map(i =>
                 axios
                     .get(`/dashboard/departmentTable/${i.DepartmentId}`)
-                    .then((data: any) => {
+                    .then((data: (string | number)[]) => {
```
Comment (kirbur) [inline src/frontend-admin/src/components/pages/DashboardPage/DashboardPage.tsx]: isn't data an array of objects?

File: src/frontend-admin/src/components/pages/HardwareDetailPage/HardwareDetailEditPage.tsx [hunk 2]
```diff
@@ -74,71 +160,544 @@ export const HardwareDetailEditPage: React.SFC<IHardwareDetailEditPageProps> = p
...
+                        )}
+                    </div>
+                )
+            }
         }
+
         return <div className={styles.section}>{rows}</div>
     }
 
+    const toggleHistoryLogForm = () => {
```
Comment (kirbur) [inline src/frontend-admin/src/components/pages/HardwareDetailPage/HardwareDetailEditPage.tsx]: this function only calls one function, why does it exist?

File: src/frontend-admin/src/components/pages/DepartmentDetailPage/DepartmentDetailEditPage.tsx [hunk 3]
```diff
@@ -63,28 +65,16 @@ export const DepartmentDetailEditPage: React.SFC<IDepartmentDetailEditPageProps>
...
 
+    //all possible hardware, software, and licenses
+    const [hardwareDropdown, setHardwareDropdown] = useState<{name: string; id: number}[]>([])
+    const [softwareDropdown, setSoftwareDropdown] = useState<{name: string; id: number}[]>([])
+    const [licenseDropdown, setLicenseDropdown] = useState<{name: string; id: number}[]>([])
+
     //input feild states:
     const [dateInput, setDateInput] = useState<Date>()
     const [deptInput, setDeptInput] = useState<{name: string; id: number}>()
     //TODO: add states for the rest of the inputs
```
Comment (kirbur) [inline src/frontend-admin/src/components/pages/DepartmentDetailPage/DepartmentDetailEditPage.tsx]: What still needs a state?

**Comment by kirbur** [comment]
A couple of console logs to remove

File: src/frontend-admin/src/components/pages/DepartmentDetailPage/DepartmentDetailEditPage.tsx (hunk 3 above)
Comment (mcdowellalex) [reply src/frontend-admin/src/components/pages/DepartmentDetailPage/DepartmentDetailEditPage.tsx]: none, lol

File: src/frontend-admin/src/components/pages/HardwareDetailPage/HardwareDetailEditPage.tsx [hunk 4]
```diff
@@ -857,7 +854,11 @@ export const HardwareDetailEditPage: React.SFC<IHardwareDetailEditPageProps> = p
                     )}
                     {!historyLogBool && (
                         <div className={styles.historyLogAdd}>
-                            <Button icon='add' onClick={toggleHistoryLogForm} className={styles.historyLogButton} />
+                            <Button
+                                icon='add'
+                                onClick={setHistoryLogBool(!historyLogBool)}
```
Comment (kirbur) [inline src/frontend-admin/src/components/pages/HardwareDetailPage/HardwareDetailEditPage.tsx]: should be onClick={() => setHistoryLogBool(!historyLogBool)}

**Pull request merged**